File: tests/uasm/sample/loop_break_continue/LoopBreakContinueTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import type { UdonInt } from "@ootr/udon-assembly-ts/stubs/UdonTypes";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class LoopBreakContinueTest extends UdonSharpBehaviour {
  Start(): void {
    let sum: UdonInt = 0n as UdonInt;

    for (let i: UdonInt = 0n as UdonInt; i < (8n as UdonInt); i++) {
      if (i === (2n as UdonInt)) {
        continue;
      }
      if (i === (6n as UdonInt)) {
        break;
      }
      sum = (sum + i) as UdonInt;
    }

    Debug.Log(sum);
  }
}

File: tests/uasm/sample/loop_break_continue/LoopBreakContinueTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class LoopBreakContinueTest : UdonSharpBehaviour
{
    public void Start()
    {
        int sum = 0;
        for (int i = 0; i < 8; i++)
        {
            if (i == 2)
            {
                continue;
            }
            if (i == 6)
            {
                break;
            }
            sum += i;
        }

        Debug.Log(sum);
    }
}
